SearchAlgoritm.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Recursive functions for the following: linear search strategy. binary search
+ * strategy.
+ * 
+ * @author Udit Saxena
+ *
+ */
+public class SearchAlgoritm {
+	/**
+	 * function to search for a given value x in an array using linear search
+	 * strategy.
+	 * 
+	 * @param array
+	 *            - integer array (sorted)
+	 * @param x
+	 *            - number to be found
+	 * @param startIndex
+	 *            - starting index of array
+	 * @return index of number or -1 if number not present
+	 */
+	int linearSearch(int[] array, int x, int startIndex) {
+		try {
+			// calling checkEmptyArray method from class QuickSortAlgorithm
+			new QuickSortAlgorithm().checkEmptyArray(array);
```

Comment:
SearchAlgoritm should not call method from QuickSortAlgorithm, This method can be moved at a common place so that both classes can use it.